libraries/AP_Airspeed/AP_Airspeed.cpp

```diff
@@ -489,7 +489,35 @@ void AP_Airspeed::init()
             num_sensors = i+1;
         }
     }
+
+#if AP_AIRSPEED_UAVCAN_ENABLED
+    // we need a 2nd pass for DroneCAN sensors so we can match order by DEVID
+    // the 2nd pass accepts any devid
+    for (uint8_t i=0; i<AIRSPEED_MAX_SENSORS; i++) {
+        if (sensor[i] == nullptr && (enum airspeed_type)param[i].type.get() == TYPE_UAVCAN) {
+            sensor[i] = AP_Airspeed_UAVCAN::probe(*this, i, 0);
+            if (sensor[i] && !sensor[i]->init()) {
+                GCS_SEND_TEXT(MAV_SEVERITY_ERROR, "Airspeed %u init failed", i + 1);
+                delete sensor[i];
```

Comment:
Is this safe if the various subscribers have been registered? I was looking at this in https://github.com/ArduPilot/ardupilot/pull/21902 and chickend out of dealocating because of the UAVCAN driver.

Reply:
the init() always returns true for DroneCAN backend, so this can't happen

Reply:
I was looking closer in any-case, there are lots of static functions that make it OK I think.